ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -778,31 +798,40 @@ private static Object extractKeyWhereClause(
     }
 
     final Expression other = getNonColumnRefSide(comparison);
-    if (!(other instanceof Literal)) {
-      throw new KsqlException("Ony comparison to literals is currently supported: " + comparison);
-    }
-
-    if (other instanceof NullLiteral) {
-      throw new KsqlException("Primary key columns can not be NULL: " + comparison);
-    }
-
-    final Object right = ((Literal) other).getValue();
-    return coerceKey(schema, right, windowed);
+    final Column keyColumn = schema.key().get(0);
+    return resolveKey(other, keyColumn, executionContext, config, comparison);
   }
 
-  private static Object coerceKey(
-      final LogicalSchema schema,
-      final Object right,
-      final boolean windowed
+  private static Object resolveKey(
+      final Expression exp,
+      final Column keyColumn,
+      final KsqlExecutionContext executionContext,
+      final KsqlConfig config,
+      final Expression errorMessageHint
   ) {
-    if (schema.key().size() != 1) {
-      throw invalidWhereClauseException("Only single KEY column supported", windowed);
+    final Object obj;
+    if (exp instanceof NullLiteral) {
+      obj = null;
+    } else if (exp instanceof Literal) {
+      // skip the GenericExpressionResolver because this is
```

Comment:
This is just done because `GenericExpressionResolver` isn't as low overhead right?

Reply:
yup, that was exactly the idea